ArcadiaTeamsBot/Dialogs/NewRequestDialog.cs

```diff
@@ -1,63 +1,288 @@
             {
-                GetInfoCard().ToAttachment()
+                return false;
+            }
+
+            if (formData[Button].ToString() == Cancel)
+            {
+                return true;
+            }
+
+            var additionalFieldsValues = new List<string>();
+            var additionalFieldsData = (formData).Properties()
+                .Where(p => p.Value.Type == JTokenType.String).ToList();
+
+            for (var i = 6; i < additionalFieldsData.Count; i++)
+            {
+                additionalFieldsValues.Add(additionalFieldsData[i].Last.ToString());
             };
-            return await stepContext.PromptAsync(nameof(TextPrompt),
-                new PromptOptions
+
+            if (string.IsNullOrEmpty(formData[Title].ToString())
+                || string.IsNullOrEmpty(formData[Description].ToString())
+                || string.IsNullOrEmpty(formData[ExecutionDate].ToString())
+                || additionalFieldsValues.Any(string.IsNullOrEmpty))
+            {
+                return false;
+            }
+
+            var newRequest = new CreateRequestDTO
+            {
+                Title = formData[Title].ToString(),
+                Description = formData[Description].ToString(),
+                Priority = Convert.ToInt32(formData[Priority].ToString()),
+                ExecutionDate = Convert.ToDateTime(formData[ExecutionDate].ToString()),
+                Username = Username,
+                FieldValues = additionalFieldsValues,
+                Type = new CreateRequestTypeDTO
+                {
+                    Id = Convert.ToInt32(formData[TypeId].ToString()),
+                    RequestTypeFields = await this.GetRequestTypeField(promptContext, cancellationToken)
+                }
+            };
+
+            var sendRequestsQuery = new CreateNewServiceDeskRequestCommand(newRequest);
+            await this.mediator.Send(sendRequestsQuery, cancellationToken);
+
+            await promptContext.Context.SendActivityAsync(MessageFactory.Text("New request created successfully"), cancellationToken);
+            return true;
+        }
+
+        private async Task<DialogTurnResult> InputStep(WaterfallStepContext stepContext, CancellationToken cancellationToken)
+        {
+            var requestTypeDTO = (ServiceDeskRequestTypeDTO)stepContext.Options;
+
+            var priorities = await this.serviceDeskClient.GetPriorities(cancellationToken);
+            var adaptiveChoiceList = priorities.Select(priority => new AdaptiveChoice { Title = priority.Value, Value = priority.Key.ToString() }).ToList();
+            
+            var cardBody = new List<AdaptiveElement>
+            {
+                new AdaptiveTextBlock { Text = requestTypeDTO.Title, Weight = AdaptiveTextWeight.Bolder, Size = AdaptiveTextSize.Large },
+
+                new AdaptiveTextInput { Id = TypeId, Value = requestTypeDTO.Id.ToString(), IsVisible = false },
+
+                new AdaptiveTextBlock(Title),
```

Comment:
You don't need to use constants everywhere or at least you need to separate constants used for ids to prevent misprints when accessing form values from constants used to show user in UI.
You use title of text block and it's input placeholder only here, you don't need to access them later from anywhere - thus you don't have to use constant.

Reply:
Changed